File: RepoTransBench/source_projects/Python/pallets_itsdangerous/public_tests/test_exc_public.py
from itsdangerous.exc import BadSignature, BadPayload, BadTimeSignature, SignatureExpired

def test_bad_signature_str_public():
    # Only the reason arg is supported in public API
    sig = BadSignature("Diff reason")
    assert "Diff reason" in str(sig)

def test_bad_payload_str_public():
    bp = BadPayload("Different")
    assert "Different" in str(bp)

def test_bad_time_signature_str_public():
    # Only message arg is guaranteed (payload/date_signed are non-API)
    bts = BadTimeSignature("ReasonZZZ")
    out = str(bts)
    assert "ReasonZZZ" in out

def test_signature_expired_str_public():
    # Only message arg is guaranteed (payload/date_signed are non-API)
    se = SignatureExpired("Late!")
    out = str(se)
    assert "Late!" in out

File: RepoTransBench/target_projects/Python/Rust/pallets_itsdangerous/public_tests/test_exc_public.rs
use itsdangerous_rs::exc::*;

#[test]
fn test_bad_signature_str_public() {
    let sig = BadSignature::new("Diff reason", None);
    assert!(format!("{}", sig).contains("Diff reason"));
}

#[test]
fn test_bad_payload_str_public() {
    let bp = BadPayload::new("Different", None);
    assert!(format!("{}", bp).contains("Different"));
}

#[test]
fn test_bad_time_signature_str_public() {
    let bts = BadTimeSignature::new("ReasonZZZ", None, None);
    let out = format!("{}", bts);
    assert!(out.contains("ReasonZZZ"));
}

#[test]
fn test_signature_expired_str_public() {
    let se = SignatureExpired::new("Late!", None, None);
    let out = format!("{}", se);
    assert!(out.contains("Late!"));
}